core/src/main/java/io/camunda/migrator/HistoryMigrator.java

```diff
@@ -415,15 +419,24 @@ private void migrateIncident(HistoricIncident legacyIncident) {
     String legacyIncidentId = legacyIncident.getId();
     if (shouldMigrate(legacyIncidentId, HISTORY_INCIDENT)) {
       HistoryMigratorLogs.migratingHistoricIncident(legacyIncidentId);
-      ProcessInstanceEntity legacyProcessInstance = findProcessInstanceByLegacyId(legacyIncident.getProcessInstanceId());
-      if (legacyProcessInstance != null) {
-        Long processInstanceKey = legacyProcessInstance.processInstanceKey();
+      ProcessInstanceEntity processInstance = findProcessInstanceByLegacyId(legacyIncident.getProcessInstanceId());
+      if (processInstance != null) {
+        Long processInstanceKey = processInstance.processInstanceKey();
         if (processInstanceKey != null) {
           Long flowNodeInstanceKey = findFlowNodeInstanceKey(legacyIncident.getActivityId(), legacyIncident.getProcessInstanceId());
           Long processDefinitionKey = findProcessDefinitionKey(legacyIncident.getProcessDefinitionId());
           Long jobDefinitionKey = null; // TODO Job table doesn't exist yet.
           IncidentDbModel dbModel = incidentConverter.apply(legacyIncident, processDefinitionKey, processInstanceKey, jobDefinitionKey, flowNodeInstanceKey);
           incidentMapper.insert(dbModel);
+          // Register incident with flow node if flow node exists
+          if (flowNodeInstanceKey != null) {
+            FlowNodeInstanceDbModel flowNodeInstance = findFlowNodeByKey(flowNodeInstanceKey);
+            if (flowNodeInstance != null) {
```

Comment:
Why don't we skip migrating the incident if the flow node instance hasn't yet been migrated?